common/app/services/mining-context.ts

```diff
@@ -1,17 +1,72 @@
-type MiningEvent = 'started-mining' | 'stopped-mining';
+type MiningEvent =
+    | 'started-mining'
+    | 'stopped-mining'
+    | 'bulk-export-started'
```

Comment:
If you move the bulk logic out of side panel I think these events and queue management will move outside of this file too.